tests/test_lib_counter.py

```diff
@@ -2,12 +2,25 @@
 This tests the files
   lib/counter.py
 """
+import mock
+from contextlib import contextmanager
 
 from .base import MyTestCase
 from privacyidea.lib.counter import increase, decrease, reset, read
 from privacyidea.models import EventCounter
 
 
+def increase_and_read(name):
+    """ helper function that increases the event counter and returns the new value """
+    increase(name)
+    return read(name)
```

Comment:
Why wouldn't we do this in ``increase`` or ``decrease``? Performance and avoiding deadlocks?

Reply:
For performance, mostly :) Also I didn't want to change all occurrences of
```r = increase(...)`` in the tests to
```
increase(...)
r = read(...)
```
so I decided to go for the helper method.